python/samples/concepts/auto_function_calling/chat_gpt_api_function_calling.py

```diff
@@ -130,13 +131,15 @@ async def handle_streaming(
 
     print("Mosscap:> ", end="")
     streamed_chunks: list[StreamingChatMessageContent] = []
-    result_content = []
+    result_content: list[StreamingChatMessageContent] = []
     async for message in response:
-        if not execution_settings.function_choice_behavior.auto_invoke_kernel_functions and isinstance(
-            message[0], StreamingChatMessageContent
+        if (
+            not execution_settings.function_choice_behavior.auto_invoke_kernel_functions
+            and isinstance(message[0], StreamingChatMessageContent)
+            and message[0].role == AuthorRole.ASSISTANT
         ):
             streamed_chunks.append(message[0])
-        else:
+        elif isinstance(message[0], StreamingChatMessageContent) and message[0].role == AuthorRole.ASSISTANT:
```

Comment:
Question: why are these conditions necessary?

Reply:
As we've been talking about -- if we simply leave the code as is, it will break. That's because all streaming CMC results are added to one list. We only want to add the Assistant message roles to this list, so that we can later form one CMC.